### ISSUE TITLE
About fp16 in training

### ISSUE THREAD

**Issue by hotcore** [IDEA-Research/detrex#143]
I have one v100 32g GPU, can i config like this"train.amp.enable=True"?
As I know v100 could use AMP(Automatic Mixed Precision)
Will fp16 accelerate training if i can use fp16?
Hope your reply~!

**Comment by rentainhe** [comment]
> I have one v100 32g GPU, can i config like this"train.amp.enable=True"? As I know v100 could use AMP(Automatic Mixed Precision) Will fp16 accelerate training if i can use fp16? Hope your reply~!

The DeformableAttention operator may not work in FP16 mode~, maybe you should run model without DeformableAttention operator

You can also try to reduce the `dataloader.train.total_batch_size` and run more iterations by setting larger `train.max_iter`

**Comment by hotcore** [comment]
Thanks for your quick reply. 
after calculate and  modify iter in train config, I can not reproduce the results of h_deformable_detr_r50_two_stage_12ep which is 48.9 in your repo. Do you have any ideas where the problem might be 
and could you please share the full training log of the 48.9 with me?

**Comment by hotcore** [comment]
> > Thanks for your quick reply. after calculate and modify iter in train config, I can not reproduce the results of h_deformable_detr_r50_two_stage_12ep which is 48.9 in your repo. Do you have any ideas where the problem might be and could you please share the full training log of the 48.9 with me?
> 
> Could you please share your training logs with us~

i will upload the logs here next week🥲appreciate❤

**Comment by rentainhe** [comment]
> > > Thanks for your quick reply. after calculate and modify iter in train config, I can not reproduce the results of h_deformable_detr_r50_two_stage_12ep which is 48.9 in your repo. Do you have any ideas where the problem might be and could you please share the full training log of the 48.9 with me?
> > 
> > 
> > Could you please share your training logs with us~
> 
> i will upload the logs here next week🥲appreciate❤

The original paper reported `48.7AP` on this, and it has some variance~ set `loss-class-weigth=1` may got better results on detrex

**Comment by hotcore** [comment]
thank you so much!!
i start next training, and wait the model performance.
could i continue to ask in this issue if i have some questions in future about hdetr？

**Comment by rentainhe** [comment]
> thank you so much!! i start next training, and wait the model performance. could i continue to ask in this issue if i have some questions in future about hdetr？

Sure~, we will keep this issue open

**Comment by hotcore** [comment]
I successfully reproduced the accuracy of the paper, thank y'all❤️❤️
excellent work！！
